lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -166,7 +170,18 @@ def upsert_records(messages)
                                   key_col_proc: key_col_proc,
                                   col_proc: col_proc,
                                   replace_associations: self.class.config[:replace_associations])
-        updater.mass_update(record_list)
+        @valid_active_records.concat(updater.mass_update(record_list))
+      end
+
+      # @param record_list [BatchRecordList]
+      # @return [Array<BatchRecord>]
+      def filter_records(record_list)
+        record_list.filter!(self.method(:should_consume?).to_proc, consume_filter)
+      end
+
+      # @return [ActiveRecord::Relation,Hash,Set]
+      def consume_filter
```

Comment:
Creates a lookup entity for use when filtering records. 
Example:
```
def consume_filter
   Widget.where(data_source: 'datafeed')
end

def should_consume?(record, filter)
   record.valid?  && filter.exlude?(record)
end
```

This is intended for use after the record is built instead of having this filtering occur in `pre_process` instead